cameraViewEx/src/main/api21/com/priyankvasa/android/cameraviewex/Camera2.kt

```diff
@@ -1160,6 +1160,14 @@ internal open class Camera2(
             }
         }
 
+        mediaRecorder?.setOnInfoListener { mediaRecorder, what, extra ->
```

Comment:
This can be set when setting up mediaRecorder ie. right below `setAudioEncoder(config.audioEncoder.value)`. Then it can be written as 
```
setAudioEncoder(config.audioEncoder.value)

setOnInfoListener { _, what, _ ->
    when (what) {
        MediaRecorder.MEDIA_RECORDER_INFO_MAX_DURATION_REACHED -> stopVideoRecording()
    }
}
```

This will avoid redundant null check `mediaRecorder?`

Reply:
Updated, much better placement.